calyx/src/analysis/live_range_analysis.rs

```diff
@@ -55,14 +31,55 @@ impl Debug for Prop {
 impl Prop {
     /// Defines the dataflow transfer function.
     /// We use the standard definition for liveness:
-    ///   `(alive - kill) + gen`
-    fn transfer(self, gen: &Prop, kill: &Prop) -> Self {
-        &(&self - kill) | gen
+    /// `(alive - kill) + gen`
+    fn transfer_in_place(&mut self, gen: Prop, kill: Prop) {
+        self.sub_in_place(kill);
```

Comment:
Any reason to have the `_in_place` suffix if there are no other methods that also do `sub` etc.